stubs/pytest.pyi

```diff
@@ -0,0 +1,14 @@
+"""
+Type stubs for pytest.
+Note that stubs are only written for the parts that we use.
+"""
+
+from typing import ContextManager, Type
+
+
+# This function actually has more arguments than are specified here, and the
+# context manager actually yields a _pytest._code.ExceptionInfo instead of None.
+# We don't use either of these features, so I don't think its worth including
+# them in our type stub. We can always change it later.
+def raises(exc_type: Type[Exception]) -> ContextManager[None]:
```

Comment:
This stub file seems entirely redundant given that the single usage of this stub is `# type: ignore`d anyway. Maybe simplify the code by dropping the stub (and ignore comment)?

Reply:
The mypy config for this project has `disallow_untyped_defs = True`, so we could remove the stub, but not the ignore comment. #31 would readd the stub again anyway, but I'm happy to remove it from this PR tomorrow if you would prefer.

Reply:
Surely it's `disallow_untyped_calls = True` which is the bit which causes the requirement here? Anyway, I'd not spotted that, so it's interesting to note.

I think if we are going to have the stubs, then let's get the signatures actually right, rather than work around our own inaccuracies. Leaving the stub incomplete seems likely to frustrate ourselves in the future.

Notably: is there a reason the stub needs to accept `Type[Exception]` rather than `Type[BaseException]`, which I expect would remove the need for the ignore?

(meta: this _is_ #31, did you mean a different PR?)

Reply:
Yeah I meant #32. 

I hadn't really looked at the stub in too much detail, I took it from https://github.com/j5api/j5/commit/86bcf3387f0e5ff1c518928d296928e8b66a8412 which was authored by @kierdavis. 

Now that I'm looking into the stubs a little more in depth, there is an unreleased upstream stub from `pytest` (https://github.com/pytest-dev/pytest/pull/5593/files#diff-fc0b2b5b62bce9258d927b9a577d6956R547). Until the stubs are released, see https://github.com/pytest-dev/pytest/issues/3342, I think we should copy the official stub. We can always remove it later. I note that they are using `BaseException`.

Reply:
e02abc8600407e459f1731a1b8124d3d2bbdc019

Reply:
I have also cherry-picked that commit across to #32